Sources/FoundationEssentials/TimeZone/TimeZone_Protocol.swift

```diff
@@ -17,6 +17,12 @@ package protocol _TimeZoneProtocol : AnyObject, Sendable, CustomDebugStringConve
     
     var identifier: String { get }
     func secondsFromGMT(for date: Date) -> Int
+
+    /// The returned `daylightSavingOffset` is different from that returned by `daylightSavingTimeOffset(:)` in that this interprets `Date` in GMT.
+    /// Essentially this is equivalent to adjusting `date` to this time zone using `rawOffset`, then passing the adjusted date to `daylightSavingTimeOffset(for: <adjusted date>)`.
+    /// This also handles the skipped time frame on DST start day differently from `daylightSavingTimeOffset(:)`, where dates in the skipped time frame are considered *not* in DST here, hence the DST offset would be 0.
+    func rawAndDaylightSavingTimeOffset(forGMTDate date: Date) -> (rawOffset: Int, daylightSavingOffset: Int)
```

Comment:
I have a naming nitpick, which you can take or leave. Calling this a "GMT" date may be confusing, since (as you well know) `Date` doesn't really have a time zone associated with it. I would just call it:

```
func rawAndDaylightSavingTimeOffset(for date: Date) -> (rawOffset: Int, daylightSavingOffset: Int)
```

Reply:
Interestingly the below `daylightSavingTimeOffset` returns a `TimeInterval`. I imagine the offset is always an integer, but it may be useful to keep that for consistency.

Reply:
I know!! I actually started with what you suggested, but I found it confusing when comparing this to the result of `daylightSavingTimeOffset(:)`. However I also agree that this current name doesn't help much disambiguating anyways, so I'm going to go back to the previous name and just leave the comment there.

Reply:
And yes `daylightSavingTimeOffset(:)` is always an integer if you look at the implementation. But that's an API so we can't change that anyways

Reply:
Sure I don't mind having it return a `TimeInterval` for the sake of consistency. We're converting it to Double anyways to add it to a `Date`